test/e2e/tracing_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+package e2e_test
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/thanos-io/thanos/pkg/promclient"
+	"github.com/thanos-io/thanos/pkg/tracing/client"
+	"github.com/thanos-io/thanos/pkg/tracing/jaeger"
+	"github.com/thanos-io/thanos/test/e2e/e2ethanos"
+
+	"github.com/efficientgo/e2e"
+	"github.com/efficientgo/tools/core/pkg/backoff"
+	"github.com/efficientgo/tools/core/pkg/testutil"
+	"gopkg.in/yaml.v2"
+)
+
+// Test to check if the trace provider works as expected.
+func TestJaegerTracing(t *testing.T) {
```

Comment:
@matej-g this is a draft of the E2E test up till now, still a WIP. 
Could you please review it up till here and let me know?
thanks!